src/Credentials/AssumeRoleWithWebIdentityCredentialProvider.php

```diff
@@ -110,7 +110,7 @@ public function __invoke()
                                 "Unreadable tokenfile at location {$this->tokenFile}"
                             );
                         }
-                        $token = file_get_contents($this->tokenFile);
+                        $token = @file_get_contents($this->tokenFile);
```

Comment:
I don't think it as such hurts to have an `@` here also, but I think it is not needed. When we get to this place, it would actually *be* and error, if there still is no token file, after the statcache has been cleared.
Unless of course you rely on fallbacks to other kind of auth, if this one fails. Then there could/should be an `@` in front also here.

Reply:
@polesen My thinking here was because we are retrying with a backoff we should silence these and allow the timeout to throw and log if we aren't able to read the credentials.

ie. We are expecting an empty token token and have a way of handling it